### PR TITLE
Updated workflow to test master branch on push

### LINKED ISSUE #396: CI/CD label must show latest status of `master` branch
## Problem
Label CD/CI shows latest status registered in Action. (Which is failing tests on PR). 

Screenshot 2020-10-13 at 17 02 47

## Expected behavior
Label CD/CI must show latest status registered in Action **of `master` branch**. (Not last registered run of PR)

### LINKED ISSUE #416: Run CD/CI pipeline against master
## Feature description

Run CD/CI pipeline against master on daily basis. 

## Why is it needed?

To show the status of the last release with the last commit in the template. The last release depends on the concrete template commit.

The template can dynamically affect vagrant-hashistack, once new commits are pushed there

## Suggestion(s)/solution(s) [Optional]

Define cron job in cd/ci workflow

## Definition of done
Tests run for the master branch on daily basis

## Checklist (after created issue)
- [X] Added label(s)
- [ ] Added to project
- [ ] Added to milestone

### PR DESCRIPTION
**Pull request by claesgill**
## Closes/fixes/resolves issue(s)?
Closes #396
Closes #416 

## What was added/changed/fixed?
Github workflow (CI/CD) test for `push` changes on `master` branch.

## Related issue(s)? [Optional]
Changelog was updated in this PR: https://github.com/fredrikhgrelland/vagrant-hashistack/pull/415

## Checklist (after created PR)
- [x] Added reviewers
- [x] Added assignee(s)
- [x] Added label(s)
- [x] Added to project
- [x] Added to milestone

### CODE REVIEW

**Comment by zhenik** [comment]
I want to point out that this PR will add a test run for the `master` branch after each merge.

The initial problem, that badge itself shows the latest status of CD/CI workflow against any branch. 

IMHO this PR closes issue #416 , but not #396

**Comment by claesgill** [comment]
> I want to point out that this PR will add a test run for the `master` branch after each merge.
> 
> The initial problem, that badge itself shows the latest status of CD/CI workflow against any branch.
> 
> IMHO this PR closes issue #416 , but not #396

You mitght be right! Let me do some testing first.

**Comment by Neha-Sinha2305** [comment]
https://github.com/fredrikhgrelland/vagrant-hashistack/blob/8b53df8df91ee841042c3bf613869f6eedfb9f6e/.github/workflows/on_pr_master.yml#L59 should be run only for event pull_request.
Since, the change from this PR will kick off the workflow for **push to master**, the workflow check will fail for **enforce_changelog action** as the action looks for **label: skip-changelog** and labels are associated only with pull requests.

File: .github/workflows/on_pr_master.yml [hunk 1]
```diff
@@ -1,8 +1,12 @@
-name: CI/CD
+name: CI-CD
```
Comment (claesgill) [inline .github/workflows/on_pr_master.yml:1]: Replaced `/` with `-` to be able to generate badge in shields.io

File: .github/workflows/on_pr_master.yml [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
 on:
+  push:
+      branches: [ master ]
   pull_request:
     branches: [ master ]
     types: [opened, synchronize, reopened, edited, closed, labeled, unlabeled]
+  schedule:
+      - cron:  '0 0 * * *'
```
Comment (claesgill) [inline .github/workflows/on_pr_master.yml:9]: Added cron job that runs at midnight: 00:00

File: README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@
 </p>
 <p align="center">
     <a href="https://github.com/fredrikhgrelland/vagrant-hashistack/actions?query=workflow%3ACI%2FCD+branch%3Amaster">
-      <img alt="Build" src="https://github.com/fredrikhgrelland/vagrant-hashistack/workflows/CI/CD/badge.svg" />
+      <img alt="Build" src="https://img.shields.io/github/workflow/status/fredrikhgrelland/vagrant-hashistack/CI-CD/master" />
```
Comment (claesgill) [inline README.md:8]: This badge do not work atm because the new changes in `on_pr_master.yml` is not merged

**Comment by zhenik** [comment]
@claesgill 
How we can test these changes?

**Comment by Neha-Sinha2305** [comment]
> https://github.com/fredrikhgrelland/vagrant-hashistack/blob/8b53df8df91ee841042c3bf613869f6eedfb9f6e/.github/workflows/on_pr_master.yml#L59
> 
> should be run only for event pull_request.
> Since, the change from this PR will kick off the workflow for **push to master**, the workflow check will fail for **enforce_changelog action** as the action looks for **label: skip-changelog** and labels are associated only with pull requests.

Code from terraform-nomad-minio:
https://github.com/fredrikhgrelland/terraform-nomad-minio/blob/master/.github/workflows/on_pr_push_master.yml#L37

**Comment by claesgill** [comment]
Have tested in my `test-git` repo: https://github.com/claesgill/test-git

> @claesgill
> How we can test these changes?

I have tested in my git repo: https://github.com/claesgill/test-git
Note that it is a failing PR and the badge is still passing.

**Review by zhenik — APPROVED** [review]

**Review by pdmthorsrud — APPROVED** [review]
LGTM

**Comment by claesgill** [comment]
> > https://github.com/fredrikhgrelland/vagrant-hashistack/blob/8b53df8df91ee841042c3bf613869f6eedfb9f6e/.github/workflows/on_pr_master.yml#L59
> > 
> > should be run only for event pull_request.
> > Since, the change from this PR will kick off the workflow for **push to master**, the workflow check will fail for **enforce_changelog action** as the action looks for **label: skip-changelog** and labels are associated only with pull requests.
> 
> Code from terraform-nomad-minio:
> https://github.com/fredrikhgrelland/terraform-nomad-minio/blob/master/.github/workflows/on_pr_push_master.yml#L37

Totally forgot to do this.. Fixed in new PR :point_right: #420

**Pull request merged**